test/e2e/workload_test.go

```diff
@@ -0,0 +1,310 @@
+			f.EventuallyRestoreSessionPhase(restoreSession.ObjectMeta).Should(Equal(v1beta1.RestoreSessionSucceeded))
+
+			By("checking the workload data has been restored")
+			restoredData, err := f.ReadDataFromMountedDirectory(deployment.ObjectMeta, framework.GetPathsFromRestoreSession(&restoreSession))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Verifying restored data is same as original data")
+			Expect(restoredData).To(BeEquivalentTo(sampleData))
+
+		})
+	})
+	Context("Leader election and backup && restore for Deployment", func() {
+		BeforeEach(func() {
+			pvc = f.GetPersistentVolumeClaim()
+			err = f.CreatePersistentVolumeClaim(pvc)
+			Expect(err).NotTo(HaveOccurred())
+			deployment = f.Deployment(pvc.Name)
+
+			targetref = v1beta1.TargetRef{
+				APIVersion: "apps/v1",
+				Kind:       apis.KindDeployment,
+				Name:       deployment.Name,
+			}
+		})
+		It("Should leader elect and Backup new deployment", func() {
+			deployment.Spec.Replicas = types.Int32P(2) // two replicas
+			By("Creating Deployment " + deployment.Name)
+			_, err = f.CreateDeployment(deployment)
+			Expect(err).NotTo(HaveOccurred())
+			err = util.WaitUntilDeploymentReady(f.KubeClient, deployment.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating sample data inside workload")
+			err = f.CreateSampleDataInsideWorkload(deployment.ObjectMeta, apis.KindDeployment)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating storage Secret " + cred.Name)
+			err = f.CreateSecret(cred)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating new repository")
+			err = f.CreateRepository(repo)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating BackupConfiguration" + backupCfg.Name)
+			err = f.CreateBackupConfiguration(backupCfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for sidecar")
+			f.EventuallyDeployment(deployment.ObjectMeta).Should(matcher.HaveSidecar(util.StashContainer))
+
+			By("Waiting for leader election")
+			f.CheckLeaderElection(deployment.ObjectMeta, apis.KindDeployment, v1beta1.ResourceKindBackupConfiguration)
+
+			By("Waiting for BackupSession")
+			f.EventuallyBackupSessionCreated(backupCfg.ObjectMeta).Should(BeTrue())
+			bs, err := f.GetBackupSession(backupCfg.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Check for repository status updated")
+			f.EventuallyRepository(&deployment).Should(WithTransform(f.BackupCountInRepositoriesStatus, BeNumerically(">=", 1)))
+
+			By("Check for succeeded BackupSession")
+			f.EventuallyBackupSessionPhase(bs.ObjectMeta).Should(Equal(v1beta1.BackupSessionSucceeded))
+
+			By("Delete BackupConfiguration")
+			err = f.DeleteBackupConfiguration(backupCfg)
+
+			By("Waiting for sidecar to be removed")
+			f.EventuallyDeployment(deployment.ObjectMeta).ShouldNot(matcher.HaveSidecar(util.StashContainer))
+
+			By("Delete sample data from workload")
+			err = f.CleanupSampleDataFromWorkload(deployment.ObjectMeta, "test-data.txt")
```

Comment:
Second parameter will be resource kind (`apis.KindDeployment`) instead of `"test-data.txt"`.